contracts/UnionZap.sol

```diff
@@ -200,7 +230,7 @@ contract UnionZap is Ownable, UnionBase {
         require(router != address(0));
         address[] memory _path = new address[](2);
         _path[0] = token;
-        _path[1] = WETH;
+        _path[1] = WETH_TOKEN;
 
         IERC20(token).safeApprove(router, 0);
```

Comment:
Setting allowance to 0 first isn't necessary since approve sets the allowance value to what's specified. There isn't risk of overflow here since the call would revert if `amount` is too large

Applying this comment to other similar cases of this throughout the contract

Reply:
Also, it looks like the method is now deprecated so use with caution https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/utils/SafeERC20.sol#L39

Reply:
It's `safeApprove` so it checks that the value is 0 first. That is because some tokens require a zero allowance before you can change the allowance (like [USDT](https://etherscan.io/address/0xdac17f958d2ee523a2206206994597c13d831ec7#code), line 199) 
`safeIncreaseAllowance` would run into that problem too, so I'd still have to `safeApprove` (or `approve` to avoid `safeApprove`) to 0 first.

Reply:
Jfc USDT

Thanks, today I learned that this monstrosity exists